CardForm/src/main/java/com/braintreepayments/api/CardAttributes.kt

```diff
@@ -0,0 +1,171 @@
+                minCardLength = 16,
+                maxCardLength = 16,
+                securityCodeLength = 3,
+                securityCodeName = R.string.bt_cvv,
+            ),
+            CardAttributes(
+                cardType = CardType.MASTERCARD,
+                pattern = Pattern.compile("^(5[1-5]|222[1-9]|22[3-9]|2[3-6]|27[0-1]|2720)\\d*"),
+                frontResource = R.drawable.bt_ic_mastercard,
+                minCardLength = 16,
+                maxCardLength = 16,
+                securityCodeLength = 3,
+                securityCodeName = R.string.bt_cvc,
+            ),
+            CardAttributes(
+                cardType = CardType.DISCOVER,
+                pattern = Pattern.compile("^(6011|65|64[4-9]|622)\\d*"),
+                frontResource = R.drawable.bt_ic_discover,
+                minCardLength = 16,
+                maxCardLength = 19,
+                securityCodeLength = 3,
+                securityCodeName = R.string.bt_cid,
+            ),
+            CardAttributes(
+                cardType = CardType.AMEX,
+                pattern = Pattern.compile("^3[47]\\d*"),
+                frontResource = R.drawable.bt_ic_amex,
+                minCardLength = 15,
+                maxCardLength = 15,
+                securityCodeLength = 4,
+                securityCodeName = R.string.bt_cid,
+            ),
+            CardAttributes(
+                cardType = CardType.DINERS_CLUB,
+                pattern = Pattern.compile("^(36|38|30[0-5])\\d*"),
+                frontResource = R.drawable.bt_ic_diners_club,
+                minCardLength = 14,
+                maxCardLength = 14,
+                securityCodeLength = 3,
+                securityCodeName = R.string.bt_cvv,
+            ),
+            CardAttributes(
+                cardType = CardType.JCB,
+                pattern = Pattern.compile("^35\\d*"),
+                frontResource = R.drawable.bt_ic_jcb,
+                minCardLength = 16,
+                maxCardLength = 16,
+                securityCodeLength = 3,
+                securityCodeName = R.string.bt_cvv,
+            ),
+            CardAttributes(
+                cardType = CardType.MAESTRO,
+                pattern = Pattern.compile("^(5018|5020|5038|5043|5[6-9]|6020|6304|6703|6759|676[1-3])\\d*"),
+                frontResource = R.drawable.bt_ic_maestro,
+                minCardLength = 12,
+                maxCardLength = 19,
+                securityCodeLength = 3,
+                securityCodeName = R.string.bt_cvc,
+                relaxedPrefixPattern = Pattern.compile("^6\\d*")
+            )
+        )
+
+        val allKnownCardBrands = knownCardBrandAttributes.values
+
+        private fun createCardAttributeMap(vararg items: CardAttributes): Map<CardType, CardAttributes> {
+            val result = mutableMapOf<CardType, CardAttributes>()
+            for (item in items) {
+                result[item.cardType] = item
+            }
+            return result
```

Comment:
Take it or leave it comment - if there's a clean/easy way to just create a map literal for `knownCardBrandAttributes`, might be nice not to need this additional helper function.

Reply:
Yeah I agree. It's a compile time vs. runtime tradeoff. I initially went for a [when statement](https://github.com/braintree/android-card-form/blob/5a355bef448b5c8e074026d9dd2699d91c219336/CardForm/src/main/java/com/braintreepayments/api/CardAttributes.kt#L71) but duplicating `CardType` felt like a code smell so I refactored to creating the map at runtime.

```kotlin
mapOf(
  CardType.VISA to CardAttributes(CardType.VISA, "^4\\d*", R.drawable.bt_ic_visa, 16, 16, 3, R.string.bt_cvv, null)
)
```